### PR TITLE
add support for redis-based rate limits on the server

### PR DESCRIPTION
**Pull request by ginkoid**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,8 +18,7 @@
   },
   "husky": {
     "hooks": {
-      "pre-commit": "yarn lint",
-      "prepare-commit-msg": "exec < /dev/tty && git cz --hook || true"
```
Comment (ethanwu10) [inline package.json]: split this out of this pr

File: test/unit/rateLimit.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    limit: 1,
+    userid
+  })
+  const result = await timeouts.checkRateLimit({
+    type: timeouts.types.UPDATE_PROFILE,
+    duration: 1000 * 10,
+    limit: 1,
+    userid
+  })
+  t.is(result.ok, false)
```
Comment (ethanwu10) [inline test/unit/rateLimit.js]: use `t.false`?

File: client/src/api/profile.js [hunk 3]
```diff
@@ -25,5 +26,12 @@ export const updateAccount = (name, division) => {
...
     division: Number.parseInt(division)
   })
-    .then(resp => resp.data)
+    .then(resp => {
+      switch (resp.kind) {
+        case 'badRateLimit':
+          toasts.useToast().add(resp.message)
+          return null
+      }
+      return resp.data
```
Comment (ethanwu10) [inline client/src/api/profile.js]: have the success case (returning data that should only be valid when the response type is what you expect) in the switch and validate type; throw error in default case?

File: server/cache/timeouts.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  end
+`)
+
+const types = {
+  FLAG: 'FLAG',
+  UPDATE_PROFILE: 'UPDATE_PROFILE'
+}
+
+module.exports = {
+  /*
```
Comment (chen-robert) [inline server/cache/timeouts.js]: we should update this comment to reflect new behavior of returning an object

**Pull request merged**